core/src/main/java/org/eclipse/hono/util/TenantObject.java

```diff
@@ -638,44 +628,26 @@ public TenantObject setTrustedCa(final Object configuration) {
     @JsonIgnore
     public Set<X500Principal> getTrustedCaSubjectDns() {
 
-        if (trustConfigurations == null) {
+        if (trustedAuthorities == null) {
             return Collections.emptySet();
         } else {
             final Set<X500Principal> subjectNames = new HashSet<>();
-            trustConfigurations.keySet().forEach(subjectDn -> {
-                subjectNames.add(new X500Principal(subjectDn));
+            trustedAuthorities.forEach(trustedCa -> {
+                subjectNames.add(new X500Principal(trustedCa.getSubjectDn()));
             });
             return subjectNames;
         }
     }
 
-    /**
-     * Get the list of configured trusted CAs for this tenant.
-     * 
-     * @return The list of configured trusted CAs or {@code null} if this tenant has no configured trusted CA.
-     *
-     */
-    @JsonIgnore
-    public List<JsonObject> getTrustedCAs() {
-
-        if (trustConfigurations == null) {
-            return null;
-        } else {
-            return trustConfigurations.values().stream().flatMap(Collection::stream).collect(Collectors.toList());
-        }
-
-    }
-
     @JsonIgnore
-    private TrustAnchor getTrustAnchor(final JsonObject trustedCa) throws GeneralSecurityException {
+    private TrustAnchor getTrustAnchor(final TrustedCertificateAuthority trustedCa) throws GeneralSecurityException {
 
-        final String subjectDn = getProperty(trustedCa, TenantConstants.FIELD_PAYLOAD_SUBJECT_DN, String.class);
-        final String encodedKey = getProperty(trustedCa, TenantConstants.FIELD_PAYLOAD_PUBLIC_KEY, String.class);
-        final String type = getProperty(trustedCa, TenantConstants.FIELD_PAYLOAD_KEY_ALGORITHM, String.class,
-                "RSA");
+        final String subjectDn = trustedCa.getSubjectDn();
+        final byte[] encodedKey = trustedCa.getPublicKey();
+        final String type = Optional.ofNullable(trustedCa.getKeyAlgorithm()).orElse("RSA");
 
-        final Instant notBefore = getNotBefore(trustedCa);
-        final Instant notAfter = getNotAfter(trustedCa);
+        final Instant notBefore = trustedCa.getNotBefore();
+        final Instant notAfter = trustedCa.getNotAfter();
         final Instant now = Instant.now();
 
         if (subjectDn == null || encodedKey == null || notBefore == null || notAfter == null) {
```

Comment:
the check against *now* means that we *always* transfer *all* CAs from the registry to the protocol adapter?

Reply:
I don't understand?

if `notBefore` is later than the current instant of time, then its early to use the CA. if now is at a later point in time than `notAfter`, then it expired. In any of these cases, a `null` trust anchor is returned. so not *all* the CAs from the registry are returned, only valid ones

Reply:
who is going to invoke this method?I believe it will be the protocol adapter, right? If that is true, then the `TenantObject` instance has been deserialized from the payload received in response to the *get Tenant* operation invocation, right? So far, we have said that the registry only transfers *currently valid* CAs to the adapters, so this filtering shouldn't be necessary, right?

Reply:
I see your point. I added this to test the `TenantObject` class: so that it can return only the valid anchors even if it contains invalid trust configurations. shall I remove it together with the corresponding tests in `TenantObjectTest`?

Reply:
I will put the check for validity inside of the `TrustedCertificateAuthority` class